tservice/src/seal.rs

```diff
@@ -90,10 +90,61 @@ impl<T: AsRef<[u8]> + core::default::Default> SealedBuilder<T> {
     }
 }
 
+pub trait Unseal<T> {
+    /// The length (in bytes) needed to hold the decrypted text
+    fn decrypted_text_len(&self) -> Result<usize>;
+
+    /// Unseal the data in `self`
+    fn unseal(&self) -> Result<T>;
+}
```

Comment:
I'm not sure I like putting the type param on this trait---it feels like it's really only capable of being used for `Vec` and constant arrays.

My preference would be to have something like this:

```rust
pub trait Unseal {
    fn decrypted_text_len(&self) -> Result<usize>;
    fn unseal(&self, dest: &mut [u8]) -> Result<usize>;

    fn unseal_vec(&self) -> Result<Vec<u8>> {
        let mut retval = vec![0u8; self.decrypted_text_len()];
        let retval_len = self.unseal(&mut retval)?;
        retval.truncate(retval_len);
        Ok(retval)
    }
}
```

The common case (give me the data as a `Vec`) is handled, but not at the expense of foreclosing the caller into using the heap or trying to guess at the size of the data from the size of the data in `Sealed<>`

Reply:
Reworked to have an `unseal` and `unseal_to_vec` method.
The trait is generic over `AsRef<[u8]>` which should match the scope of `Sealed`